openapi.yaml

```diff
@@ -0,0 +1,712 @@
+      operationId: getServices
+      responses:
+        '200':
+          description: Successful operation _(service list)_
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Service'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too many requests
+      security:
+        - subscription_key: []
+  /services/{serviceId}:
+    get:
+      tags:
+        - services
+      summary: Retrieve service
+      description: Retrieve a service by `serviceId`
+      operationId: getService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation _(service data)_
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+      security:
+        - subscription_key: []
+    put:
+      tags:
+        - services
+      summary: Update service
+      description: |
+        Update an existing service by by `serviceId`. <br><br>_Note: This API operation needs the __Manage API key__ owned by the service user, otherwise `403 forbidden` will be returned to the caller._
+      operationId: updateService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Updated service payload
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServicePayload'
+        required: true
+      responses:
+        '200':
+          description: Successful operation _(service updated)_
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service cannot be updated)_
+      security:
+        - subscription_key: []
+    delete:
+      tags:
+        - services
+      summary: Delete service
+      description: |
+        Delete a service by `serviceId`. <br>This operation is a logical deletion: the service persist in user services, but it's status become
```

Comment:
I agree with @rodoviti, and I add a question (possibly not related with this PR): what if I `getService` on a deleted item? Would it be:
1. 404 Not Found
2. 200 { status: DELETED }

Reply:
@balanza for me should return a 404 Not Found, but it might depend on the Use Cases.
I also add that this operation could (or should?!) be idempotent

Reply:
_404_ should be more user "transparent", but they loose the possibility of knowing if a service is deleted or not and in this case _200 { status: DELETED }_ is a good opportunity.
We have to choose...

Reply:
What do you think about using the `423 Locked` code to distinguish a really not found service _(404)_ from a canceled one _(423)_?

> The HTTP 423 Locked error response code indicates that either the resources tentatively targeted by is locked, meaning it can't be accessed.

Reply:
**_Final decision:_** `getService` of a deleted item will return `200` and the related service with `status=deleted`